cf/commands/servicekey/service_key_test.go

```diff
@@ -0,0 +1,125 @@
+	var (
+		ui                  *testterm.FakeUI
+		config              core_config.Repository
+		cmd                 ServiceKey
+		requirementsFactory *testreq.FakeReqFactory
+		serviceRepo         *testapi.FakeServiceRepo
+		serviceKeyRepo      *testapi.FakeServiceKeyRepo
+	)
+
+	BeforeEach(func() {
+		ui = &testterm.FakeUI{}
+		config = testconfig.NewRepositoryWithDefaults()
+		serviceRepo = &testapi.FakeServiceRepo{}
+		serviceInstance := models.ServiceInstance{}
+		serviceInstance.Guid = "fake-service-instance-guid"
+		serviceRepo.FindInstanceByNameMap = generic.NewMap()
+		serviceRepo.FindInstanceByNameMap.Set("fake-service-instance", serviceInstance)
+		serviceKeyRepo = testapi.NewFakeServiceKeyRepo()
+		cmd = NewGetServiceKey(ui, config, serviceRepo, serviceKeyRepo)
+		requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, TargetedSpaceSuccess: true, ServiceInstanceNotFound: false}
+	})
+
+	var callGetServiceKey = func(args []string) bool {
+		return testcmd.RunCommand(cmd, args, requirementsFactory)
+	}
+
+	Describe("requirements", func() {
+		It("fails when not logged in", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: false}
+			Expect(callGetServiceKey([]string{"fake-service-key-name"})).To(BeFalse())
+		})
+
+		It("requires two arguments and one option to run", func() {
+			Expect(callGetServiceKey([]string{})).To(BeFalse())
+			Expect(callGetServiceKey([]string{"fake-arg-one"})).To(BeFalse())
+			Expect(callGetServiceKey([]string{"fake-arg-one", "fake-arg-two", "fake-arg-three"})).To(BeFalse())
+		})
+
+		It("fails when service instance is not found", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, ServiceInstanceNotFound: true}
+			Expect(callGetServiceKey([]string{"non-exist-service-instance"})).To(BeFalse())
+		})
+
+		It("fails when space is not targetted", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, TargetedSpaceSuccess: false}
+			Expect(callGetServiceKey([]string{"fake-service-instance", "fake-service-key-name"})).To(BeFalse())
+		})
+	})
+
+	Describe("requiremnts are satisfied", func() {
+		Context("get service key successfully", func() {
+			BeforeEach(func() {
+				serviceKeyRepo.GetServiceKeyMethod.ServiceKey = models.ServiceKey{
+					Fields: models.ServiceKeyFields{
+						Name:                "fake-service-key",
+						Guid:                "fake-service-key-guid",
+						Url:                 "fake-service-key-url",
+						ServiceInstanceGuid: "fake-service-instance-guid",
+						ServiceInstanceUrl:  "fake-service-instance-url",
+					},
+					Credentials: map[string]interface{}{
+						"username": "fake-username",
+						"password": "fake-password",
+						"host":     "fake-host",
+						"port":     "3306",
+						"database": "fake-db-name",
+						"uri":      "mysql://fake-user:fake-password@fake-host:3306/fake-db-name",
+					},
+				}
+			})
+
+			It("get service credential", func() {
```

Comment:
Minor: should read `It("gets service ...")` same also for rest. Reember the tests description should almost read like English proses... as much as you can. I know this is not super important but since we are there and I saw it, I am pointting them out...

Finally, I realize that English is not first language so some of these typos/grammar errors are normal. Cheers.